### PR TITLE
[Kotlin][Client] Add support for JodaTime

### PR DESCRIPTION
**Pull request by javils**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [X] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [X] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### PR Description
Add support for joda time library. I know that it's a not supported library as you can see [here](https://github.com/JodaOrg/joda-time) but there are a lot of java project using this library and they are very coupled to this library. 

So I think that we can offer a easy transition for these project supporting JodaTime and they could migrate to kotlin and after that, remove these dependencies if they want.

@jimschubert @dr4ke616 @karismann @Zomzog @andrewemery @4brunu

### CODE REVIEW

File: samples/client/petstore/kotlin-jodatime/src/main/kotlin/org/openapitools/client/infrastructure/LocalDateAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package org.openapitools.client.infrastructure
+
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.ToJson
+import java.time.LocalDate
```
Comment (4brunu) [inline samples/client/petstore/kotlin-jodatime/src/main/kotlin/org/openapitools/client/infrastructure/LocalDateAdapter.kt]: @javils Those imports are wrong, they should be from joda time

File: samples/client/petstore/kotlin-jodatime/src/main/kotlin/org/openapitools/client/infrastructure/LocalDateTimeAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package org.openapitools.client.infrastructure
+
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.ToJson
+import java.time.LocalDateTime
```
Comment (4brunu) [inline samples/client/petstore/kotlin-jodatime/src/main/kotlin/org/openapitools/client/infrastructure/LocalDateTimeAdapter.kt]: @javils Those imports are wrong, they should be from joda time

**Comment by jimschubert** [comment]
Has this PR been abandoned?

@4brunu should we add support for Joda via separate PR? I've never found a need for Joda, so I don't know how popular it is or why one would prefer this over Java time. We also support template customization with minimal effort, so I'd personally prefer having users make that simple customization rather than implement joda, threetenp, etc across all JVM generators.

**Comment by 4brunu** [comment]
@jimschubert I don't need the support for JodaTime. I was just trying to help with the code review 🙂 
I think the only reason to add support for JodaTime is in legacy projects.

**Comment by wing328** [comment]
@javils did you have time to review the feedback? Is this PR still needed?

**Comment by jimschubert** [comment]
@wing328 I think the PR can't be merged until the imports are fixed anyway. Adding JodaTime support as another lib to maintain when Java Time API is now preferred seems to add more maintenance overhead which hasn't had a lot of community request and which is easily addressed using custom templates.

**Comment by jimschubert** [comment]
Haven't heard back in regards to the discussion here.
I think rather than incorporating this into Kotlin, users who need it can use existing functionality (import/type mappings, or template customization) to achieve this.

**Pull request closed without merging**